data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -77,7 +77,6 @@ public PipelineModel(
             @JsonProperty("sink") final List<SinkModel> sinks,
             @JsonProperty("workers") final Integer workers,
             @JsonProperty("delay") final Integer delay) {
-        checkArgument(Objects.nonNull(source), "Source must not be null");
```

Comment:
Why is source not required anymore? Doesn't even DLQ pipeline have a `pipeline` source?

Reply:
@graytaylor0 DLQ pipelines do not have source because source/processor/buffer/sink can end events to DLQ pipeline. So, the source is the new "HeadlessPipelineSource" that I added. This source is created automatically for a DLQ pipeline. It is not configurable.